### PR TITLE
fix: 修复 React 18 环境下 Tooltip 卸载后无法再次渲染 & 排序菜单选中效果丢失的问题

### PR DESCRIPTION
**Pull request by lijinke666**
### 👀 PR includes

🐛 Bugfix

- [x] Solve the issue and close #0

### 📝 Description

image

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| image | image |
|image | image |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/src/sheet-type/spread-sheet.ts [hunk 1]
```diff
@@ -281,14 +279,7 @@ export abstract class SpreadSheet extends EE {
         onMounted: resolve,
       };
 
-      if (isMobile()) {
```
Comment (lijinke666) [inline packages/s2-core/src/sheet-type/spread-sheet.ts:284]: 之前移动端的抽屉, 有部分代码没删除干净

File: packages/s2-react/src/components/tooltip/components/operator.tsx [hunk 2]
```diff
@@ -65,7 +65,7 @@ export const TooltipOperator: React.FC<Required<TooltipOperatorProps>> =
           mode={onlyShowOperator ? 'vertical' : 'horizontal'}
           className={cls(`${TOOLTIP_PREFIX_CLS}-operator-menus`, className)}
           onClick={onMenuClick}
-          defaultSelectedKeys={defaultSelectedKeys}
```
Comment (lijinke666) [inline packages/s2-react/src/components/tooltip/components/operator.tsx:68]: defaultSelectedKeys 导致的选中状态未更新

File: packages/s2-react/src/components/tooltip/custom-tooltip.tsx [hunk 3]
```diff
@@ -60,27 +51,11 @@ export class CustomTooltip extends BaseTooltip<
       this.forceClearContent();
     }
 
-    const Content = this.isMobileDevice() ? (
```
Comment (lijinke666) [inline packages/s2-react/src/components/tooltip/custom-tooltip.tsx:63]: 无用代码

File: packages/s2-react/src/components/tooltip/custom-tooltip.tsx [hunk 4]
```diff
@@ -125,6 +100,8 @@ export class CustomTooltip extends BaseTooltip<
     // https://github.com/facebook/react/issues/25675#issuecomment-1363957941
     Promise.resolve().then(() => {
       this.root?.unmount();
+      // Fiber 节点卸载后不能再重新渲染, 需要重新创建
+      this.root = null;
```
Comment (lijinke666) [inline packages/s2-react/src/components/tooltip/custom-tooltip.tsx:104]: image

React 18 unmount 后, 不能再 render 了, 需要创建一个新的 root

**Review by wjgogogo — APPROVED** [review]

**Pull request merged**